avm/ptn/azd/container-app/main.bicep

```diff
@@ -0,0 +1,249 @@
+metadata name = 'avm/ptn/azd/container-app'
```

Comment:
The PR looks good to go @NanaXiong00.
I'd just have one question: Do you know why we happen to have 2 patterns that are named always the same?
```bicep
metadata name = 'avm/ptn/azd/container-apps'
metadata description = 'Creates an Azure Container Registry and an Azure Container Apps environment.'
```
and
```bicep
metadata name = 'avm/ptn/azd/container-app'
metadata description = 'Creates a container app in an Azure Container App environment.'
```

Namewise - I can't help but wonder if it's not mildly confusing 😄 I assume there's a common team responsible for the `azd` modules, so I'd just be curious to know if you feel like these 2 modules are, name- & description-wise, distinct enough.

Want to make sure we got this covered before either of the two are merged. ([ref](https://github.com/Azure/bicep-registry-modules/blob/2eeb04162c7f26fddf66bf5b7b401895c85cf6fe/avm/ptn/azd/container-apps/main.bicep))

Reply:
@AlexanderSehr The above two ptn module names are named according to the two files [container-apps.bicep](https://github.com/Azure/azure-dev/blob/main/templates/common/infra/bicep/core/host/container-apps.bicep) and [container-app.bicep](https://github.com/Azure/azure-dev/blob/main/templates/common/infra/bicep/core/host/container-app.bicep) in azure-dev. These names have been reviewed and confirmed by @jongio.

Reply:
Fair and maybe it's just me that would consider this naming very very 'unfortunate'. @jongio, any guess as to why this happened? 
Wouldn't a name like `/azd/registry-linked-container-app` made more sense for this pattern, and `/azd/container-app-environment` for the other?

I'm not going to block your PRs over it, but given that these are published to the public registry for everybody to use, I at least wanted to bring it up.

Reply:
@jongio What do you think of AlexanderSehr's suggestion above?

Reply:
@wbreza - Thoughts on if we should use this opportunity to rename them?

Reply:
If we want to do it, we should do it now. Technically nothing is published and just the entry in the MAR file would need to be replaced & the GitHub teams would need to be renamed.

Reply:
Hey @jongio @NanaXiong00 did you have the chance to discuss the above renaming suggestion?

Reply:
> Hey @jongio @NanaXiong00 did you have the chance to discuss the above renaming suggestion?

@jongio About the renaming suggestion, any ideas?

Reply:
Agree that now is the time to rename them since they are so similar.  The `container-apps` module is really all about creating the prereqs and dependencies required by the actual container apps.  Perhaps we should consider using one of the following:

- container-apps-stack
- container-apps-setup
- container-apps-foundation
- container-apps-prereqs

Reply:
@matebarabas, any thoughts? This would require us to 
- update the AVM issues for the modules
- create a new PR to the MAR file, updating the original values
- updating the PR(s) to reflect the new telemetry value, folder naming etc.

Just simply updating the name for both the 
- azd/container-app and
- azd/container-apps
modules as per whatever name is prefered for either one.

Not sure who wants to ultimately decide the name, but I think @wbreza's suggestions sound already a lot better :)

Reply:
I'm fully supportive of renaming modules that don't have a meaningful enough name. In fact, I've already called out exactly this problem in the internal planning thread, but that conversation concluded on keeping the original names for those modules that were eventually (two of them got ditched in the process).

So, please let me know what's the new name for which one of these, and I'm happy to take the necessary steps (most of them highlighted by Alex above):
- `avm/ptn/azd/container-app` ---> to be renamed to ??? OR are we keeping this and just renaming the below one?
- `avm/ptn/azd/container-apps` ---> to be renamed to ???

Internally, I've also asked about this one:
- `avm/ptn/azd/container-app-upsert`---> to be renamed to ???

Thanks!

Reply:
I'd strongly recommend to rename both, as neither describes the corresponding module's content. Moreover, both are essentially just called the same which doesn't help.

Reply:
100% agree with Alex here.

Reply:
Thanks, @wbreza! Do you need any other input or approval from other folks in the team, or shall we continue with the renaming to the above? Thanks.
CC: @jongio

Reply:
ship it

Reply:
Excellent! Let me take the next steps. I'll let everyone know (here), when everything is ready.

Reply:
Checklist:
- [x] update the AVM module proposal issues and BRM PRs with the new module names
- [x] update the module indexes with the new names
- [x] submit a PR to update the MAR manifest file with the new names
- [ ] MAR manifest PR merge confirmed (**pending, module PRs cannot be published until this step is complete!**)
- [ ] rename the related GH teams @jongio to grant me maintainer access on each of the below teams, so I can help:
  - [avm-ptn-azd-containerapp-module-contributors-bicep · Azure Team (github.com)](https://github.com/orgs/Azure/teams/avm-ptn-azd-containerapp-module-contributors-bicep)
  - [avm-ptn-azd-containerapp-module-owners-bicep · Azure Team (github.com)](https://github.com/orgs/Azure/teams/avm-ptn-azd-containerapp-module-owners-bicep)
  - [avm-ptn-azd-containerapps-module-contributors-bicep · Azure Team (github.com)](https://github.com/orgs/Azure/teams/avm-ptn-azd-containerapps-module-contributors-bicep)
  - [avm-ptn-azd-containerapps-module-owners-bicep · Azure Team (github.com)](https://github.com/orgs/Azure/teams/avm-ptn-azd-containerapps-module-owners-bicep)
- [ ] update both PR(s) to reflect the new telemetry value, folder naming etc.
  - [ ] https://github.com/Azure/bicep-registry-modules/pull/3234
  - [ ] https://github.com/Azure/bicep-registry-modules/pull/3221

Reply:
Done

Reply:
@jongio, you only added me as a member in each, can you please promote me to a maintainer so that I have rights to make changes? Thanks!

You need to check the box next to my name and then select change role: 
![image](https://github.com/user-attachments/assets/e14bf0c0-2a24-4de8-a7a6-718bf71d507c)

Then, select maintainer:
![image](https://github.com/user-attachments/assets/5af003c3-8a99-47aa-a437-4320c0a2fb53)

Reply:
Thanks, Jon, for granting me access. I'm done with the edits and have removed myself from the teams.

Reply:
Resolving